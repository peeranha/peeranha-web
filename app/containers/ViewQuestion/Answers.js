import React from 'react';
import PropTypes from 'prop-types';

import AnswersTitle from './AnswersTitle';
import AnswersList from './AnswersList';
import AcceptedAnswer from './AcceptedAnswer';

const Answers = props => (
  <div className="answers">
    <AnswersTitle
      translations={props.translations}
      answersNum={props.questionData.answers.length}
    />
    <AcceptedAnswer {...props} />
    <AnswersList {...props} />
  </div>
);

Answers.propTypes = {
  questionData: PropTypes.object,
  translations: PropTypes.object,
};

export default Answers;
