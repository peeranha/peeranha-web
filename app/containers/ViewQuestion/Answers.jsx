import React from 'react';
import PropTypes from 'prop-types';

import AnswersTitle from './AnswersTitle';
import AnswersList from './AnswersList';
import AcceptedAnswer from './AcceptedAnswer';

export const Answers = props => {
  const answersList = props.questionData.answers;

  if (!answersList.length) return null;

  return (
    <div>
      <AnswersTitle answersNum={answersList.length} />

      {answersList[0].content && (
        <React.Fragment>
          <AcceptedAnswer {...props} />
          <AnswersList {...props} />
        </React.Fragment>
      )}
    </div>
  );
};

Answers.propTypes = {
  questionData: PropTypes.object,
};

export default React.memo(Answers);
