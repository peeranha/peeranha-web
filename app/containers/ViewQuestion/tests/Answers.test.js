import { Answers } from '../Answers';

describe('Answers', () => {
  it('test', () => {
    const props = {
      questionData: {
        answers: [],
      },
    };
    expect(Answers(props)).toMatchSnapshot();
  });
});
