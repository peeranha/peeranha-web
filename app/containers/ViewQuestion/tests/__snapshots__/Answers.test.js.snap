// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers test 1`] = `
<div
  className="answers"
>
  <AnswersTitle
    answersNum={0}
  />
  <AcceptedAnswer
    questionData={
      Object {
        "answers": Array [],
      }
    }
  />
  <AnswersList
    questionData={
      Object {
        "answers": Array [],
      }
    }
  />
</div>
`;
